Add glossary to linting (#113)
exercism/docs: building/configlet/lint.md

- The `"contributors"` values must not have duplicates
- The `"contributors"` values are treated case-insensitively
- Users can only be listed in either the `"authors"` or `"contributors"` array (no overlap)

## Glossary

> - _kebab_case_: strings which characters are all in the range [a-z09-]

Reviewer: ```suggestion
- _kebab_case_: strings containing only characters in the range `[a-z0-9-]`
```

> - _kebab_case_: strings containing only characters in the range `[a-z0-9-]`

Author: ```suggestion
- _kebab_case_: strings containing only characters in the range `[a-z0-9-]`; the first character must be in the range `[a-z0-9]` (no dash)

Author: @ee7 This clear?

Reviewer 2: Depends on whether you want `hello------------erik` and `hello-` to be valid slugs :)

Reviewer: I think we should go with the regexp`^([a-z0-9][a-z0-9]*)(-[a-z0-9]+)*$`

Author: ```suggestion
- _kebab_case_: strings starting and ending with a character in the range `[a-z0-9]`, with zero or more characters in between in the range `[a-z0-9-]`. In regular expression format: `^[a-z0-9]+(-[a-z0-9]+)*$`
```

Author: @iHiD @ee7 Something like this?

Reviewer: ```suggestion
- _kebab_case_: strings containing only characters in the range `[a-z0-9]` optionally separated by dashes (e.g. "two-fer"). In regular expression format: `^[a-z0-9]+(-[a-z0-9]+)*$`
```
